File: app/src/main/java/dev/pitlor/smssync/activities/fragments/SettingsFragment.java
package dev.pitlor.smssync.activities.fragments;

import android.os.Bundle;

import androidx.preference.PreferenceFragmentCompat;

import dagger.hilt.android.AndroidEntryPoint;
import dev.pitlor.smssync.R;

@AndroidEntryPoint
public class SettingsFragment extends PreferenceFragmentCompat {
    @Override
    public void onCreatePreferences(Bundle savedInstanceState, String rootKey) {
        setPreferencesFromResource(R.xml.settings, rootKey);
    }
}

File: app/src/main/java/dev/pitlor/smssync/activities/fragments/SettingsFragment.kt
package dev.pitlor.smssync.activities.fragments

import android.os.Bundle
import androidx.preference.PreferenceFragmentCompat
import dagger.hilt.android.AndroidEntryPoint
import dev.pitlor.smssync.R

@AndroidEntryPoint
class SettingsFragment : PreferenceFragmentCompat() {
    override fun onCreatePreferences(savedInstanceState: Bundle, rootKey: String) {
        setPreferencesFromResource(R.xml.settings, rootKey)
    }
}